tests/test_taskFactory.py

```diff
@@ -0,0 +1,162 @@
+    initInput = connectionTypes.InitInput(name="fakeInitInput", doc="", storageClass="SourceCatalog")
+    initOutput = connectionTypes.InitOutput(name="fakeInitOutput", doc="", storageClass="SourceCatalog")
+    input = connectionTypes.Input(name="fakeInput", doc="", storageClass="SourceCatalog", dimensions=set())
+    output = connectionTypes.Output(name="fakeOutput", doc="", storageClass="SourceCatalog", dimensions=set())
+
+
+class FakeConfig(PipelineTaskConfig, pipelineConnections=FakeConnections):
+    widget = pexConfig.Field(dtype=float, doc="")
+
+
+def mockTaskClass():
+    """A class placeholder that records calls to __call__.
+    """
+    mock = unittest.mock.Mock(__name__="_TaskMock", _DefaultName="FakeTask", ConfigClass=FakeConfig)
+    return mock
+
+
+class TaskFactoryTestCase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        tmp = tempfile.mkdtemp()
+        cls.addClassCleanup(shutil.rmtree, tmp, ignore_errors=True)
+        cls.repo = butlerTests.makeTestRepo(tmp)
+        butlerTests.addDatasetType(cls.repo, "fakeInitInput", set(), "SourceCatalog")
+        butlerTests.addDatasetType(cls.repo, "fakeInitOutput", set(), "SourceCatalog")
+        butlerTests.addDatasetType(cls.repo, "fakeInput", set(), "SourceCatalog")
+        butlerTests.addDatasetType(cls.repo, "fakeOutput", set(), "SourceCatalog")
+
+    def setUp(self):
+        super().setUp()
+
+        self.factory = TaskFactory()
+        self.tracker = mockTaskClass()
+
+    @staticmethod
+    def _alteredConfig():
+        config = FakeConfig()
+        config.widget = 42.0
+        return config
+
+    @staticmethod
+    def _overrides():
+        overrides = ConfigOverrides()
+        overrides.addValueOverride("widget", -1.0)
+        return overrides
+
+    @staticmethod
+    def _dummyCatalog():
+        schema = afwTable.SourceTable.makeMinimalSchema()
+        return afwTable.SourceCatalog(schema)
+
+    def _tempButler(self):
+        butler = butlerTests.makeTestCollection(self.repo, uniqueId=self.id())
+        catalog = self._dummyCatalog()
+        butler.put(catalog, "fakeInitInput")
+        butler.put(catalog, "fakeInitOutput")
+        butler.put(catalog, "fakeInput")
+        butler.put(catalog, "fakeOutput")
+        return butler
+
+    def testNoOptions(self):
+        self.factory.makeTask(taskClass=self.tracker, label=None, config=None,
+                              overrides=None, butler=None)
+        self.tracker.assert_called_with(config=FakeConfig(), initInputs=None, name=None)
+
+    def testAllOptions(self):
+        butler = self._tempButler()
+        self.factory.makeTask(taskClass=self.tracker, label="no-name", config=self._alteredConfig(),
+                              overrides=self._overrides(), butler=butler)
+        catalog = butler.get("fakeInitInput")  # Copies of _dummyCatalog are identical but not equal
+        # Overrides ignored
```

Comment:
Please expand on this comment: I think you mean "when `config` is passed, `overrides` is ignored", but I'm not sure.